### PR TITLE
Essay sorting

### LINKED ISSUE #21: Admin clicks on any application
- They will be shown the essay and all total scores for that application.
- Each criteria will be averaged across all reviews.
- A list of each review with criteria scores will be shown to the side.
- Admins will have access to Name and Email of student.
- An admin can not edit any scores submitted by a reviewer.

### PR DESCRIPTION
**Pull request by icorson3**
@srt32 @VictoriaVasys:

- Adds id to admin application view since this is what the reviewers will see, makes it easy to match up app id to user if there are problems. 
- Removes unnecessary `self` calls in `admin/dashboard` controller.
- Adds an id and state scope to cohort.applications

Closes #21

### CODE REVIEW

File: app/models/cohort.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Cohort < ApplicationRecord
-  has_many :applications, dependent: :destroy
+  has_many :applications, -> { order('state DESC, id') }, dependent: :destroy
```
Comment (srt32) [inline app/models/cohort.rb]: I wonder if we could put the sorting logic in the controller layer so we don't sort everywhere we make this call. WDYT?

Feel free to punt

**Comment by srt32** [comment]
:+1:

File: app/models/cohort.rb (hunk 1 above)
Comment (icorson3) [reply app/models/cohort.rb]: @srt32 I actually was thinking of a better solution as I did that for that reason and I think i came up with something that works and is just for this view. Let me know what you think

File: app/models/cohort.rb [hunk 2]
```diff
@@ -12,26 +12,30 @@ class Cohort < ApplicationRecord
   scope :current, -> { where('open_date <= ? AND close_date >= ?', Date.today, Date.today) }
   scope :closed, -> { where('open_date > ? OR close_date < ?', Date.today, Date.today) }
 
+  def order_by_id_and_state
+    applications.order('state DESC, id')
```
Comment (srt32) [inline app/models/cohort.rb]: :+1: to pulling out a method here. This method doesn't need to be a class method?

Comment (srt32) [reply app/models/cohort.rb]: never mind! looks good.

File: app/controllers/admin/dashboard_controller.rb [hunk 3]
```diff
@@ -1,10 +1,10 @@
 class Admin::DashboardController < AdminController
 
   def index
-    @cohorts = Cohort.all.to_json(:include => [{:applications => {:include => [:user, :reviews]}}, :reviewers, :non_reviewers], methods: [:formatted_close_date, :formatted_start_date, :formatted_notify_date])
+    @cohorts = Cohort.all.to_json(:include => [{:applications => {:include => [:user, :reviews]}}, :reviewers, :non_reviewers, :order_by_id_and_state => {:include => [:user, :reviews]}], methods: [:formatted_close_date, :formatted_start_date, :formatted_notify_date])
```
Comment (srt32) [inline app/controllers/admin/dashboard_controller.rb]: What does this change do? Are we not already including applications? What if we sorted client-side? This blob of data is getting crazy!

File: app/models/cohort.rb [hunk 4]
```diff
@@ -12,26 +12,30 @@ class Cohort < ApplicationRecord
   scope :current, -> { where('open_date <= ? AND close_date >= ?', Date.today, Date.today) }
   scope :closed, -> { where('open_date > ? OR close_date < ?', Date.today, Date.today) }
 
+  def order_by_id_and_state
```
Comment (srt32) [inline app/models/cohort.rb]: I think I'd name this method something like `applications_ordered_by_blahblah` to make it clear it deals with apps as opposed to sorting cohorts

File: app/controllers/admin/dashboard_controller.rb (hunk 3 above)
Comment (icorson3) [reply app/controllers/admin/dashboard_controller.rb]: Yeah, I agree but I was more going with the app to add the features for now before refactoring but it seems that a serializer would do something similar.

File: app/models/cohort.rb (hunk 4 above)
Comment (icorson3) [reply app/models/cohort.rb]: 👍

**Comment by icorson3** [comment]
@srt32 updated the method name

**Comment by srt32** [comment]
🎉

**Comment by srt32** [comment]
@icorson3 good to merge here?

**Comment by srt32** [comment]
Fantastic! Please delete your branches once merged as well.

**Pull request merged**